### PR TITLE
Button: center icon when icon only

### PR DESCRIPTION
**Pull request by breville**
This updates the `Button` component so that if an icon is provided with no accompanying text, the icon is now centered.  Previously, the empty text element would still be rendered, creating unnecessary space to the right of the icon.

### before
Screenshot 2023-11-16 at 4 03 25 PM

### after

Screenshot 2023-11-16 at 4 06 48 PM

### CODE REVIEW

**Review by bencodeorg — APPROVED** [review]
Seems reasonable to me -- you feel confident that we don't keep this span around even when its empty for some legitimate reason (eg, can be helpful with flexbox layouts sometimes)?

**Comment by breville** [comment]
> Seems reasonable to me -- you feel confident that we don't keep this span around even when its empty for some legitimate reason (eg, can be helpful with flexbox layouts sometimes)?

It's a great question.  One reason I extracted this into its own PR is that it could have broad impact on our UI.  I can't think of a specific scenario in which this would actually break things, though I can imagine some Eyes tests might be affected as buttons slightly resize.

**Comment by bencodeorg** [comment]
FWIW, no sign of flexbox usage in the button SCSS file or the component itself

File: apps/src/templates/Button.jsx [hunk 1]
```diff
@@ -206,7 +206,9 @@ class Button extends React.Component {
               <FontAwesome icon="spinner" className="fa-spin" />
             </span>
           )}
-          <span className={moduleStyles.textSpan}>{!isPending && text}</span>
+          {!isPending && text && (
+            <span className={moduleStyles.textSpan}>{!isPending && text}</span>
```
Comment (levadadenys) [inline apps/src/templates/Button.jsx]: nit: if compiler goes to this line of code, it already passes this if, so `!isPending && text` can be simplified to `text`

**Review by levadadenys — APPROVED** [review]
Thanks a lot for this fix!

File: apps/src/templates/Button.jsx (hunk 1 above)
Comment (breville) [reply apps/src/templates/Button.jsx]: Good point, fixed!

**Pull request closed without merging**